jupyter-book/appendix/glossary.md

```diff
@@ -6,6 +6,14 @@ of this page.
 
 ## Main terms used in this course
 
+### API
+
+Refers to both the specific interfaces for estimators implemented in
+scikit-learn and the generalized conventions across types of estimators.
+
+In scikit-learn we try to adopt simple conventions and limit to a minimum the
+number of methods an object must implement.
+
```

Comment:
```suggestion 

Note that, more generally, "API" is an acronym that stands for "Application
Programming Interface". It can have a slightly different meaning in different
contexts: in some cases it can be used to designate an online service that can be
accessed by remote programs. In the context of an online service, the term "API"
can be used both to designate the service itself, and the technical specification
of the programming interface used by people who write client applications that
connect to this service.

In the context of an (offline) library such as scikit-learn, it just means the
list of all (public) functions, classes and methods in the library, along with the
specification of the accepted argument types.
```